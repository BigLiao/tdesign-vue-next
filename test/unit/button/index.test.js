import { mount } from "@vue/test-utils";
import Button from "@/button";

describe("Button", () => {
  let cmp;

  beforeEach(() => {
    cmp = mount(Button,{
      propsData: {
        round: true
      }
    })
  });

  it("equals round to true", () => {
    expect(cmp.vm.round).toEqual(true);
  });
});