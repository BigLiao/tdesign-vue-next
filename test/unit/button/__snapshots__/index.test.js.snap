// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button :props :block 1`] = `<button class="t-button t-button--line t-size-full-width">text</button>`;

exports[`Button :props :disabled 1`] = `<button class="t-button t-button--line t-is-disabled">text</button>`;

exports[`Button :props :icon function 1`] = `<button class="t-button t-button--line t-button--icon-only"><i>custom icon</i></button>`;

exports[`Button :props :icon string 1`] = `<button class="t-button t-button--line"><i class="t-icon t-icon-delete"></i><span class="t-button__text">text</span></button>`;

exports[`Button :props :loading 1`] = `<button class="t-button t-button--line t-is-loading"><i class="t-icon t-icon-loading"></i><span class="t-button__text">text</span></button>`;

exports[`Button :props :size 1`] = `<button class="t-button t-size-l t-button--line">text</button>`;

exports[`Button :props :theme 1`] = `<button class="t-button t-button--primary">text</button>`;

exports[`Button :props :theme 2`] = `<button class="t-button t-button--line" round="true">text</button>`;

exports[`Button <slot> <icon> 1`] = `
<button class="t-button t-button--line t-button--icon-only">
  <div></div>
</button>
`;
